Pull requests from non-collaborator forked repositories failed due missing env var

# Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [X] I am running the latest version
> Running latest version of develop tagged on `e7a4b6a`
- [X] I checked the documentation and found no answer
> No clear location for documentation other than the README 😢 , opening that as a new issue.
- [X] I checked to make sure that this issue has not already been filed
> ✅ 
- [X] I'm reporting the issue to the correct repository.
> Indeed, this is related to `pleuston` repository missing an environment variable.

## Expected Behavior

>Please describe the behavior you are expecting.
A pull-request should be able to grab a `Token` environment variable from Travis.

## Current Behavior

>What is the current behavior? Please provide as many details as possible.
See #158. Builds fail as it's missing the variable, as can be seen [here](https://travis-ci.com/oceanprotocol/pleuston/jobs/156160539).

## Step to Reproduce the problem

>Please provide detailed steps for reproducing the issue.

1. Create a pull request as non-collaborator from a forked repository
2. Watch the build complain about the missing token 

### Possible Solution

Issue seems to appear in non-collaborator pull requests, so there are a couple ways to solve it:

- Create a set of specific jobs in Travis that can pass secrets and use a development account so no production secrets are passed.
- Copy this pull request's branch and push it to upstream with one of your collaborators.
- Provide some information in how to include this information locally or through Travis.

## Specifications

>Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

No required for this issue.

## Failure Logs

>Please include any relevant log snippets or files here.
```
[error] "2018-11-03T18:43:59.471Z"  'Error sending coverage'
[error] "2018-11-03T18:43:59.473Z"  Error: Token is required
    at Object.module.exports [as handleInput] (/home/travis/build/oceanprotocol/pleuston/node_modules/codacy-coverage/lib/handleInput.js:22:35)
    at Socket.<anonymous> (/home/travis/build/oceanprotocol/pleuston/node_modules/codacy-coverage/bin/codacy-coverage.js:48:20)
    at emitNone (events.js:111:20)
    at Socket.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:139:11)
    at process._tickCallback (internal/process/next_tick.js:181:9)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @oceanprotocol/pleuston@0.1.0 test: `npm run lint && node scripts/test.js --env=jsdom --coverage && cat ./coverage/lcov.info | codacy-coverage`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @oceanprotocol/pleuston@0.1.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2018-11-03T18_43_59_489Z-debug.log
The command "npm run test" exited with 1.
27.83s$ npm run build
```
More can be seen in the following failed Travis build - https://travis-ci.com/oceanprotocol/pleuston/jobs/156160539

--- a/.travis.yml
+++ b/.travis.yml
@@ -12,7 +12,8 @@ install:
 - npm install
 
 script:
-- npm run test:ci
+- npm test
+- 'if [ "$TRAVIS_SECURE_ENV_VARS" = "true" ]; then npm run coverage; fi'
 - npm run build
 
 notifications:

--- a/README.md
+++ b/README.md
@@ -23,19 +23,21 @@ Main issues right now:
 
 ## Table of Contents
 
-- [Features](#features)
-- [Prerequisites](#prerequisites)
-    - [🐋 aquarius](#aquarius)
-    - [💧 keeper-contracts](#keeper-contracts)
-- [Storage Providers](#storage-providers)
-    - [Azure Storage](#azure-storage)
-- [Development](#development)
-    - [MetaMask](#metamask)
-    - [Production build](#production-build)
-- [Configuration](#configuration)
-- [Code style](#code-style)
-- [Testing](#testing)
-- [License](#license)
+- [Pleuston](#pleuston)
+    - [Table of Contents](#table-of-contents)
+    - [Features](#features)
+    - [Prerequisites](#prerequisites)
+        - [🐋 aquarius](#-aquarius)
+        - [💧 keeper-contracts](#-keeper-contracts)
+    - [Storage Providers](#storage-providers)
+        - [Azure Storage](#azure-storage)
+    - [Development](#development)
+        - [MetaMask](#metamask)
+        - [Production build](#production-build)
+    - [Configuration](#configuration)
+    - [Testing](#testing)
+    - [Code style](#code-style)
+    - [License](#license)
 
 ---
 
@@ -149,21 +151,26 @@ module.exports = {
 }
 ```
 
+## Testing
+
+Automatic tests are setup via Travis, executing `npm test`.
+
+The tests run:
+
+- linting checks with ESLint and Stylelint
+- basic rendering tests for components with Jest
+
+While code coverage status will be reported in the console and on Travis, coverage information won't be uploaded to Codacy for any Pull Request from a forked repo. That is because of a [security restriction in Travis](https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions).
+
 ## Code style
 
 Code linting is setup with [ESLint](https://eslint.org) and [stylelint](https://stylelint.io) following [eslint-config-oceanprotocol](https://github.com/oceanprotocol/eslint-config-oceanprotocol) and [stylelint-config-bigchaindb](https://github.com/bigchaindb/stylelint-config-bigchaindb).
 
-There's a npm script setup which runs all linting tests:
+There's a npm script setup which runs only linting tests:
 
 ```bash
 npm run lint
 ```
-
-## Testing
-
-Automatic tests are setup via Travis, executing `npm test:ci`. If you want to execute these tests locally, please use `npm test`. Otherwise, setup an account with [Codacy](https://www.codacy.com/) and provide an API Token as described in their [documentation](https://support.codacy.com/hc/en-us/articles/115000255385).
-
-At the moment, besides linting tests, there's only one test checking if the whole app can be rendered.
 
 ## License
 

--- a/package.json
+++ b/package.json
@@ -10,7 +10,7 @@
     "lint:css": "stylelint ./src/**/*.scss",
     "lint": "run-p --continue-on-error lint:js lint:css",
     "test": "npm run lint && node scripts/test.js --env=jsdom --coverage",
-    "test:ci": "npm run test && cat ./coverage/lcov.info | codacy-coverage",
+    "coverage": "cat ./coverage/lcov.info | codacy-coverage",
     "requirements-check": "./scripts/check-node-version.sh",
     "preinstall": "npm run requirements-check"
   },

